### PR TITLE
🔐 OIDC back-channel logout

### LINKED ISSUE #509: Implement OIDC back-channel-logout
As describe in [TWP linagora/twake-workplace-private#99](https://github.com/linagora/twake-workplace-private/issues/99), we want to have logout. The only unbuggy way to do it is to use OIDC [Back Channel Logout](https://openid.net/specs/openid-connect-backchannel-1_0.html) like Twake-Mail and Twake-Chat.

Please implement it _(with [backchannel_logout_session_required](https://openid.net/specs/openid-connect-backchannel-1_0.html#BCRegistration), to avoid deleting all sessions)_.

### PR DESCRIPTION
**Pull request by MontaGhanmy**

### CODE REVIEW

File: tdrive/backend/node/src/services/console/clients/remote-jwks-verifier.ts [hunk 1]
```diff
@@ -97,4 +97,12 @@ export class OidcJwtVerifier {
 
     return jwt;
   }
+
+  async verifyLogoutToken(logoutTokenString: string) {
+    const jwt = await this.verifyAsPromise(logoutTokenString);
+    // verifyAudience(expectedClientId, jwt.claims.aud);
```
Comment (ericlinagora) [inline tdrive/backend/node/src/services/console/clients/remote-jwks-verifier.ts:103]: are the comments left here for future use ?

File: tdrive/backend/node/test/e2e/console/backchannel-logout.spec.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  beforeEach(async () => {
+    platform = await init();
+  });
+
+  afterEach(async () => {
+    await platform.tearDown();
+    platform = null;
+  });
+
+  beforeAll(async () => {
```
Comment (ericlinagora) [inline tdrive/backend/node/test/e2e/console/backchannel-logout.spec.ts:17]: why is the init call in both beforeEach and beforeAll ?

File: tdrive/backend/node/src/services/console/entities/session.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export const TYPE = "session";
+
+@Entity(TYPE, {
+  primaryKey: [["sub"]],
+  type: TYPE,
+})
+export default class Session {
+  @Column("sub", "string")
+  sub: string;
```
Comment (ericlinagora) [inline tdrive/backend/node/src/services/console/entities/session.ts:16]: could be worth documenting that this is a userId ?

**Review by ericlinagora — APPROVED** [review]

File: tdrive/backend/node/src/core/platform/services/database/services/orm/utils.ts [hunk 4]
```diff
@@ -110,5 +110,5 @@ export function toMongoDbOrderable(timeuuid?: string): string {
  * @returns {boolean} Returns true if key is "is_in_trash" or "scope", otherwise returns false.
  */
 export const filteringRequired = (key: string) => {
-  return key === "is_in_trash" || key === "scope";
+  return key === "is_in_trash" || key === "scope" || "sub";
```
Comment (ericlinagora) [inline tdrive/backend/node/src/core/platform/services/database/services/orm/utils.ts]: `key === "sub"` no ? needs doc comment update too i think

Comment (MontaGhanmy) [reply tdrive/backend/node/src/core/platform/services/database/services/orm/utils.ts]: Yes @ericlinagora , thank you.

**Pull request merged**